### PR TITLE
refactor: Convert AbstractSchedTest to Kotlin 

### PR DESCRIPTION
**Pull request by david-allison**

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/libanki/sched/AbstractSchedTest.kt [hunk 1]
```diff
@@ -299,146 +337,193 @@ mw.col.sched.extendLimits(1, 0)
 ```
              */
         }
+
+        init {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/sched/AbstractSchedTest.kt:341]: @KotlinCleanup("remove init and initialize the properties directly")

File: AnkiDroid/src/test/java/com/ichi2/libanki/sched/AbstractSchedTest.kt [hunk 2]
```diff
@@ -299,146 +337,193 @@ mw.col.sched.extendLimits(1, 0)
...
-        d.put("name", name);
-        decks.update(d);
+        val did = addDeck(name)
+        val d = decks.get(did)
+        d.put("name", name)
+        decks.update(d)
 
-        boolean hasMatch = decks.all().stream().anyMatch(x -> name.equals(x.getString("name")));
-        assertThat(String.format("Deck %s should exist", name), hasMatch, is(true));
+        val hasMatch = decks.all().stream().anyMatch { x: Deck -> name == x.getString("name") }
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/sched/AbstractSchedTest.kt:451]: @KotlinCleanup("Replace stream() with kotlin collection operators")

**Review by lukstbit — CHANGES_REQUESTED** [review]
Looks ok with two extra cleanups.

**Review by lukstbit — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
nice - looks good
passes local regression check, let's go

**Pull request merged**